geode-pulse/src/test/java/org/apache/geode/tools/pulse/tests/Server.java

```diff
@@ -103,26 +160,63 @@ public Server(int jmxPort, String properties, String jsonAuthFile) throws Except
     }
 
     cs.start();
+    Awaitility.waitAtMost(30, TimeUnit.SECONDS).pollInterval(1, TimeUnit.SECONDS)
+        .until(() -> cs.isActive() == true);
   }
 
-  private String formJMXServiceURLString(String host, int jmxPort) throws UnknownHostException {
-    String jmxSerURL = "";
+  private synchronized void unloadMBeans() throws InstanceNotFoundException {
+    JMXProperties props = JMXProperties.getInstance();
+    try {
+      props.load(propFile);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
 
-    InetAddress inetAddr = InetAddress.getByName(host);
-    if (inetAddr instanceof Inet4Address) {
-      // Create jmx service url for IPv4 address
-      jmxSerURL = "service:jmx:rmi://" + host + "/jndi/rmi://" + host + ":" + jmxPort + "/jmxrmi";
-    } else if (inetAddr instanceof Inet6Address) {
-      // Create jmx service url for IPv6 address
-      jmxSerURL =
-          "service:jmx:rmi://[" + host + "]/jndi/rmi://[" + host + "]:" + jmxPort + "/jmxrmi";
+    // remove servers
+    String[] servers = getArrayProperty(props, "servers");
+    for (String server : servers) {
+      try {
+        removeServerMBean(server);
+      } catch (MBeanRegistrationException e) {
+        e.printStackTrace();
+      } catch (MalformedObjectNameException e) {
+        e.printStackTrace();
+      } catch (NullPointerException e) {
+        e.printStackTrace();
+      }
     }
 
-    return jmxSerURL;
-  }
+    // remove members
+    String[] members = getArrayProperty(props, "members");
+    for (String m : members) {
+      try {
+        removeMemberMBean(m);
+      } catch (InstanceAlreadyExistsException e) {
+        e.printStackTrace();
+      } catch (MBeanRegistrationException e) {
+        e.printStackTrace();
+      } catch (NotCompliantMBeanException e) {
+        e.printStackTrace();
+      } catch (MalformedObjectNameException e) {
+        e.printStackTrace();
+      } catch (NullPointerException e) {
```

Comment:
Do we really want to swallow NullPointerExceptions and just print them?  Seems like at very least we should wrap this and the above exceptions in a TestException so it fails the test.  Or just add them to the exceptions thrown by this method.

Reply:
good feedback. It was based on old code. I have removed all of that.